core/aws-core/src/main/java/software/amazon/awssdk/awscore/internal/client/config/AwsClientOptionValidation.java

```diff
@@ -39,8 +39,8 @@ public static void validateSyncClientOptions(SdkClientConfiguration c) {
     }
 
     private static void validateClientOptions(SdkClientConfiguration c) {
-        require("credentialsProvider", c.option(AwsClientOption.CREDENTIALS_PROVIDER));
```

Comment:
Again, I think we need to fall back to this if CREDENTIALS_IDENTITY_PROVIDER is null.

Reply:
Confirmed in offline discussion - fallback not needed.